esports_data_from_rimble/rimble/rimble_test.go

```diff
@@ -0,0 +1,286 @@
+package rimble_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"testing"
+	"time"
+
+	lom "github.com/samber/lo/mutable"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/blocky/attestation-service-examples/esports-data-from-rimble/rimble"
+)
+
+var RIMBLE_DEMO_API_KEY = "TU167z1Pwb9SAbUErPZN2aepia1MOsBN3nXbC1eE"
+
+func fetchRawMatchData(date, matchID, apiKey string) (rimble.MatchData, error) {
+	url := fmt.Sprintf("https://rimbleanalytics.com/raw/csgo/match-status/?matchid=%s&date=%s", matchID, date)
+
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return rimble.MatchData{}, fmt.Errorf("error creating request: %w", err)
+	}
+
+	req.Header.Add("x-api-key", apiKey)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return rimble.MatchData{}, fmt.Errorf("error making request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return rimble.MatchData{}, fmt.Errorf("API request failed with status code: %d, URL: %s", resp.StatusCode, req.URL.String())
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return rimble.MatchData{}, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	match, err := rimble.MakeMatchDataFromMatchesJSON(body)
+	if err != nil {
+		err = fmt.Errorf(
+			"making match given data '%s' and match ID '%s': %w",
+			date,
+			matchID,
+			err,
+		)
+		return rimble.MatchData{}, err
+	}
+
+	return match, nil
+}
+
+func TestMatchData_Winner(t *testing.T) {
```

Comment:
There is more testing that we could do on these functions, but I don't think we need to go overboard. Mostly I wanted to have something to check that the `lo` functions were doing stuff correctly after I reimplemented this example after using nested loops initially to extract the data